autogen/oai/mistral.py

```diff
@@ -126,26 +153,31 @@ def parse_params(self, params: Dict[str, Any]) -> Dict[str, Any]:
         # This can occur when using LLM summarisation
         for i in range(1, len(mistral_messages)):
             if mistral_messages[i - 1].role == "assistant" and mistral_messages[i].role == "system":
```

Comment:
I think this iteration is repetitive and can be merge with loop above. Also, why this code here? Is it because  system message can not exist after the first message, or system message can not be after assistant message?

Reply:
Good pickup, I'll move it into the above loop.

Yes, it will throw an exception if a system message is after an assistant message (role order).

Reply:
I'll mark as resolved with the latest commit, I've moved it into the above loop.